srcalc/src/test/java/gov/va/med/srcalc/vista/RpcVistaPatientDaoTest.java

```diff
@@ -67,4 +77,53 @@ public final void testSaveNoteSuccess() throws Exception
     	        SaveNoteCode.SUCCESS,
     	        dao.saveRiskCalculationNote(PATIENT_DFN, ELECTRONIC_SIGNATURE, DUMMY_BODY));
     }
+    
+    @Test
+    public final void testLabRetrievalFailure() throws Exception
+    {
+        final List<String> labNames = new ArrayList<String>();
+        labNames.add("ALBUMIN");
+        final VistaProcedureCaller caller = mock(VistaProcedureCaller.class);
+        rpcSetup(caller);
+        // Anything besides a valid measurement is returned as an empty string
+        when(caller.doRetrieveLabs(eq(RADIOLOGIST_DUZ), eq(String.valueOf(PATIENT_DFN)), anyListOf(String.class)))
+            .thenReturn("");
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        // Behavior verification
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        assertEquals(new HashMap<String, RetrievedValue>(), patient.getLabs());
+    }
+    
+    @Test
+    public final void testLabRetrievalSuccess() throws Exception
+    {
+        final List<String> labNames = new ArrayList<String>();
+        labNames.add("ALBUMIN");
+        final VistaProcedureCaller caller = mock(VistaProcedureCaller.class);
+        rpcSetup(caller);
+        when(caller.doRetrieveLabs(RADIOLOGIST_DUZ, String.valueOf(PATIENT_DFN), labNames))
+            .thenReturn(ALBUMIN_SUCCESS);
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        // Behavior verification
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        assertEquals(1, patient.getLabs().size());
+        final RetrievedValue value = patient.getLabs().get("ALBUMIN");
+        assertEquals(3.0, value.getValue(), .0001);
+        final Calendar cal = Calendar.getInstance();
+        cal.set(2015, 1, 2, 14, 35, 12);
+        cal.set(Calendar.MILLISECOND, 0);
+        assertEquals(cal.getTime(), value.getMeasureDate());
+    }
+    
+    private static void rpcSetup(final VistaProcedureCaller caller)
```

Comment:
Good idea to extract out this code. Let's go further and make this a true factory method that takes no arguments and returns a mock `VistaProcedureCaller`. All these tests can then standardize on using that method. See `DefaultAdminServiceTest` for an example.